modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala

```diff
@@ -111,28 +123,49 @@ trait PropPathBuilder {
       }
     }
 
+    def many(paths: List[Path]): Either[PropPathError, PropPath] = {
+      Try {
```

Comment:
No need to `Try` and throw, you can directly foldleft over eithers and compose them with flatMap (`import cats.syntax.either._`)

Reply:
But I want to stop the fold loop whenever I see the first error. One error in any of the paths will mean the whole thing fails. And throwing scapes from the fold loop